efficientdet/aug/mosaic.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2020 Google Research. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from functools import partial
+import tensorflow as tf
+from typing import Tuple
+from absl import logging
+
+
+class Mosaic:
+  """Mosaic.
+        Mosaic Augmentation class.
+        Notes:- 1. Mosaic sub images will not be preserving aspect ratio of original images passed.
+                2. Tested on static graphs and eager  execution.
+                3. This Implementation of mosaic augmentation is tested in tf2.x.
+    """
+
+  def __init__(
+      self,
+      out_size: Tuple = (680, 680),
+      n_images: int = 4,  #Currently four images are supported!
+      _minimum_mosaic_image_dim: int = 25,
+  ):
+    """__init__.
+
+        Args:
+            out_size: output mosaic image size.
+            n_images: number images to make mosaic
+            _minimum_mosaic_image_dim: minimum percentage of out_size dimension should the mosaic be. i.e if out_size is (680,680) and _minimum_mosaic_image_dim is 25 , minimum mosaic sub images dimension will be 25 % of 680
```

Comment:
This line seems too long.

Could you use yafs to format your code: https://github.com/google/automl/blob/master/efficientdet/g3doc/faq.md#21-how-can-i-format-my-code-for-prs

Reply:
yapf --in-place --style='{based_on_style=yapf}' aug/mosaic.py

Reply:
i did use this

Reply:
can you format from your end? we can squash  the commits later.

Reply:
Oh, I think because it is in a comment start with """, then the formater couldn't reformat it.  Could you just manually make them fit "80 characters per line" rule? (should be fairly simple.

Reply:
sure.

Reply:
@mingxingtan Resolved!. Thanks for pointing out, we can merge it now.